reporting/report.go

```diff
@@ -27,6 +28,27 @@ func Init() *Report {
 	}
 }
 
+func (r *Report) AddSecret(finding secrets.Finding) {
+	done := false
+
+	// If secret already exists, just add the link of new finding
+	if len(r.Results[finding.ID]) > 0 {
+		for i, s := range r.Results[finding.ID] {
+			if s.Value == finding.Secret {
+				s := &r.Results[finding.ID][i]
+				s.Links = append(s.Links, finding.Source)
+				done = true
+				break
+			}
+		}
+	}
+	// If secret don't exist on a specific source or if source doesn't exist just create the secret
+	if !done {
+		secret := Secret{ID: finding.ID, Links: []string{finding.Source}, Description: finding.Description, StartLine: finding.StartLine, EndLine: finding.EndLine, StartColumn: finding.StartColumn, EndColumn: finding.EndColumn, Value: finding.Secret}
+		r.Results[finding.ID] = append(r.Results[finding.ID], secret)
```

Comment:
the ID for a confluence page might be unique, I don't know if it is, but for the file system the ID will only be the name of the file, that is not unique, the source is the one that is unique since it represents the fullpath (or the relative path if we want to change the logic)